### PR TITLE
feat: support passing in session token

### LINKED ISSUE #9: Support passing in a session token with credentials
### Issues

- [X] I have checked [existing issues](https://github.com/RimacTechnology/semantic-release-ecr/issues) and there are no existing ones with the same request.

### Feature description

Thanks for making this plugin!

I'm using the official AWS action to authenticate my workflows, but I'm doing so through the OIDC flow via GitHub. Even though this is [the recommended authentication flow ](https://github.com/aws-actions/configure-aws-credentials?tab=readme-ov-file#assumerolewithwebidentity-recommended), it's not supported by this plugin. Specifically, the AWS access ID and secret key generated through this action is associated with an AWS session token. If this session token isn't passed, the credentials won't work.

In the `aws.ts` file, we create an `ECRClient`:
```typescript
constructor(accessKeyId: string, region: string, secretAccessKey: string) {
    this.awsEcr = new ECRClient({
        credentials: {
            accessKeyId,
            secretAccessKey,
        },
        region,
    })
}
```

The `credentials` object also supports a `sessionToken` property. I think it should be as simple as just passing it along via an environment variable, but keep it optional for those who don't need it.

I opened pull request #8 as a starting point for the changes. Is this something we could get through?

### PR DESCRIPTION
**Pull request by makmn1**
Adds support for passing in a session token associated with an AWS access id and secret key pair. For example, when assuming a role through GitHub's OIDC flow for AWS, the generated access id and secret are linked to the session token, and won't work unless the session token is also passed

### CODE REVIEW

**Review by mario-jerkovic — APPROVED** [review]
Thanks for opening this PR, @makmn1! 

Really appreciate the time you took to contribute this feature! 🙌

File: yarn.lock
Comment (mario-jerkovic) [inline yarn.lock:1]: @makmn1 this file shouldn't change. It looks like you have used different version of yarn to do `yarn install` can you please remove/reset this file on this PR as no dep. has changes

Comment (makmn1) [reply yarn.lock:1]: Done!

**Pull request merged**